### PR TITLE
issue #26156 Prevent debits and credits to the same account cancelling out

### PR DESCRIPTION
**Pull request by ChristopherCotnoir**

### CODE REVIEW

**Comment by rkannadevara** [comment]
Unable to perform UAT: Please provide more information to verify the fix.

**Comment by ChristopherCotnoir** [comment]
If you create an open item such that the offsetting account is the same account, it should be posted to the GL as a credit and a debit to the same account.

File: foundation-database/public/functions/postglseries.sql [hunk 1]
```diff
@@ -106,13 +106,14 @@ BEGIN
   FOR _glseries IN SELECT glseries_source, glseries_doctype, glseries_docnumber,
                           glseries_accnt_id, glseries_distdate, glseries_notes,
                           glseries_misc_id,
-                          SUM(glseries_amount) as amount
+                          SUM(glseries_amount) as amount,
+                          ROUND(glseries_amount/ABS(glseries_amount), 0) as credit
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/postglseries.sql]: This can be reduced to a simple boolean check: `(glseries_amount > 0) as credit`

**Comment by sindhupriyaav** [comment]
Thanks for the information. Performed UAT: Issue is fixed.

**Pull request merged**